typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -2289,7 +2289,7 @@
 [special-comment-value (-> -Special-Comment Univ)]
 
 ;; Section 13.8
-[prop:custom-write -Struct-Type-Property]
+[prop:custom-write (-struct-property (-> -Self -Output-Port B ManyUniv))]
```

Comment:
The third argument in this function type should not be `B` / `Boolean`. It should allow a boolean, `0`, or `1`.

The [documentation for `prop:custom-write`](https://docs.racket-lang.org/reference/Printer_Extension.html#(def._((quote._~23~25kernel)._prop~3acustom-write))) through the [documentation for `gen:custom-write`](https://docs.racket-lang.org/reference/Printer_Extension.html#%28def._%28%28lib._racket%2Fprivate%2Fbase..rkt%29._gen~3acustom-write%29%29) specifies that it

> takes three arguments: the structure to be printed, the target port, and an argument that is `#t` for write mode, `#f` for display mode, or `0` or `1` indicating the current quoting depth for print mode.

This seems to specify the type `(U #t #f 0 1)` equivalently `(U Boolean 0 1)` for that third argument.

Reply:
You can construct the `(U #t #f 0 1)` type with `(Un -True -False -Zero -One)` for the "base-env" declaration.